tests/widget/widget.js

```diff
@@ -110,4 +112,85 @@ describe( 'Widget', () => {
 		sinon.assert.notCalled( focusSpy );
 		expect( getModelData( document ) ).to.equal( '[<widget></widget>]' );
 	} );
+
+	describe( 'delete and backspace handling', () => {
```

Comment:
There are far more false positive cases than not. And the risk of this code handling too many cases is, in fact, much worse for the editor than not handling some widget's case, because it may break the general editor behaviour. So when adding some extension to already tested mechanism, we must be sure that this extension is very precisely targetted. It's really more important than checking the positive cases.